main/test/ca/uwaterloo/flix/language/phase/TestEntryPoint.scala

```diff
@@ -0,0 +1,118 @@
+    val input =
+      """
+        |def main(blah: Array[Char]): Int32 & Impure = ??? as & Impure
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.UnexpectedEntryPointArg](result)
+  }
+
+  test("Test.UnexpectedEntryPointArg.Main.02") {
+    val input =
+      """
+        |def main(blah: Array[a]): Int32 & Impure = ??? as & Impure
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.UnexpectedEntryPointArg](result)
+  }
+
+  test("Test.UnexpectedEntryPointArg.Main.03") {
+    val input =
+      """
+        |class C[a]
+        |
+        |def main(blah: Array[a]): Int32 & Impure with C[a] = ??? as & Impure
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.UnexpectedEntryPointArg](result)
+  }
+
+  test("Test.TooManyEntryPointArgs.Main.01") {
+    val input =
+      """
+        |def main(arg1: Array[String], arg2: Array[String]): Unit = ???
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.TooManyEntryPointArgs](result)
+  }
+
+  test("Test.TooManyEntryPointArgs.Main.02") {
+    val input =
+      """
+        |def main(arg1: String, arg2: String): Unit = ???
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.TooManyEntryPointArgs](result)
+  }
+
+  test("Test.UnexpectedEntryPointResult.Main.01") {
+    val input =
+      """
+        |def main(blah: Array[String]): a & Impure = ??? as & Impure
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.UnexpectedEntryPointResult](result)
+  }
+
+  test("Test.UnexpectedEntryPointResult.Main.02") {
+    val input =
+      """
+        |enum E
+        |def main(blah: Array[String]): E = ???
+        |""".stripMargin
+    val result = compile(input, Options.TestWithLibMin)
+    expectError[EntryPointError.UnexpectedEntryPointResult](result)
+  }
+
+  //  test("Test.IllegalMain.07") { // MATT ok
+  //    val input =
+  //      """
+  //        |def main(blah: Array[String]): Int32 & ef = ???
+  //        |""".stripMargin
+  //    val result = compile(input, Options.TestWithLibMin)
+  //  }
+  //
+  //  test("Test.IllegalMain.01") { // MATT move to happy test
+  //    val input =
+  //      """
+  //        |def main(blah: Array[String]): Int32 = ???
+  //        |""".stripMargin
+  //    val result = compile(input, Options.TestWithLibMin)
+  //    expectError[TypeError.IllegalMain](result)
+  //  }
+  //
+  //  test("Test.IllegalMain.03") { // MATT ok
+  //    val input =
+  //      """
+  //        |def main(blah: Array[String]): Int64 & Impure = ???
+  //        |""".stripMargin
+  //    val result = compile(input, Options.TestWithLibMin)
+  //    expectError[TypeError.IllegalMain](result)
+  //  }
+
+
```

Comment:
Do we want a few positive tests somewhere? I think can think of these:

```
def main(): Unit ==
def main(): Int32 ==
def main(): String =
def main(_args: Array[String]): Unit =
def main(_args: Array[String]): Int32 =
def main(_args: Array[String]): String =
```

Reply:
We can. It's just slightly burdensome since each will need its own file. I guess we can just put them here instead of in `*.flix` files.